BoxContentSDK/BoxContentSDK/Requests/BOXFolderRequest.m

```diff
@@ -59,64 +80,140 @@ - (BOXAPIOperation *)createOperation
+                [JSONoperation.APIRequest addValue:notMatchingEtag
+                                forHTTPHeaderField:BOXAPIHTTPHeaderIfNoneMatch];
+            }
         }
+        
+        [self addSharedLinkHeaderToRequest:JSONoperation.APIRequest];
+        
+        return JSONoperation;
     }
-
-    [self addSharedLinkHeaderToRequest:JSONoperation.APIRequest];
-
-    return JSONoperation;
 }
 
 - (void)performRequestWithCompletion:(BOXFolderBlock)completionBlock
 {
     if (completionBlock) {
-        BOOL isMainThread = [NSThread isMainThread];
-        BOXAPIJSONOperation *folderOperation = (BOXAPIJSONOperation *)self.operation;
-
-        folderOperation.success = ^(NSURLRequest *request, NSHTTPURLResponse *response, NSDictionary *JSONDictionary) {
-            BOXFolder *folder = [[BOXFolder alloc] initWithJSON:JSONDictionary];
-
-            [self.sharedLinkHeadersHelper storeHeadersFromAncestorsIfNecessaryForItemWithID:folder.modelID
+        if (self.associateId != nil) {
+            BOOL isMainThread = [NSThread isMainThread];
+            
+            BOXAPIDataOperation *folderOperation = (BOXAPIDataOperation *)self.operation;
+            
+            __weak BOXAPIDataOperation *weakFileOperation = folderOperation;
+            
+            folderOperation.successBlock = ^(NSString *modelID, long long expectedTotalBytes) {
+                NSData *data = [NSData dataWithContentsOfFile:weakFileOperation.destinationPath];
+                BOXFolder *folder = nil;
+                if (data != nil) {
+                    NSDictionary *jsonDictionary = [NSJSONSerialization JSONObjectWithData:data options:kNilOptions error:nil];
+                    folder = (BOXFolder*)[[self class] itemWithJSON:jsonDictionary];
+                    
+                    [self.sharedLinkHeadersHelper storeHeadersFromAncestorsIfNecessaryForItemWithID:folder.modelID
+                                                                                           itemType:folder.type
+                                                                                          ancestors:folder.pathFolders];
+                    
+                    if ([self.cacheClient respondsToSelector:@selector(cacheFolderRequest:withFolder:error:)]) {
+                        [self.cacheClient cacheFolderRequest:self
+                                                  withFolder:folder
+                                                       error:nil];
+                    }
+                }
+                
+                if ([[NSFileManager defaultManager] fileExistsAtPath: weakFileOperation.destinationPath]) {
+                    NSError *error = nil;
+                    [[NSFileManager defaultManager] removeItemAtPath:weakFileOperation.destinationPath error:&error];
+                }
+                
+                [BOXDispatchHelper callCompletionBlock:^{
+                    completionBlock(folder, nil);
```

Comment:
mangled output would be folder == nil? Yeah, we talked briefly. We could return a content sdk type of error for not being able to parse output or so.